roles/ceph-mon/tasks/crush_rules.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: create roots needed for configured crush rules
+  command: "{{ docker_exec_cmd }} ceph --cluster {{ cluster }} osd crush add-bucket {{ item.root }} root"
+  with_items: "{{ crush_rules | unique }}"
+  changed_when: false
+  failed_when: false
+  run_once: true
+
+- name: create configured crush rules
+  command: "{{ docker_exec_cmd }} ceph --cluster {{ cluster }} osd crush rule create-simple {{ item.name }} {{ item.root }} {{ item.type }}"
+  with_items: "{{ crush_rules | unique }}"
+  changed_when: false
+  failed_when: false
+  run_once: true
+
+- name: get id for new default crush rule
+  command: "{{ docker_exec_cmd }} ceph --cluster {{ cluster }} osd -f json crush rule dump {{ item.name }}"
+  register: info_ceph_default_crush_rule
+  changed_when: false
+  failed_when: false
+  with_items: "{{ crush_rules }}"
+  when: item.default
+
+- name: set crush rule info as fact
+  set_fact:
+    info_ceph_default_crush_rule_yaml: "{{ info_ceph_default_crush_rule.results[0].stdout|from_json() }}"
+  when: info_ceph_default_crush_rule.results|length > 0
+
+# Continue with common tasks
+- name: add new default crush rule to ceph.conf
```

Comment:
I'd be nice to avoid the restart here, can we try to inject the option into the mon instead?
Like this  `ceph daemon mon.devstack config set osd_pool_default_crush_replicated_ruleset <id>`.
Then apply the new config option into `ceph.conf` but just don't restart the mon.
You'd have to apply that on all the mons obviously.

I just tried and it seems to work fine.